### ISSUE TITLE
Clarification Needed: Indexing Difference for 'ans_logits' and 'ans_tokens'

### ISSUE THREAD

**Issue by HZD01** [apartresearch/Integer_Addition#2]
### Description:

In the `logits_to_tokens_loss` function, `ans_logits` and `ans_tokens` are indexed differently, but the comments suggest they should refer to the same tokens. Here’s the relevant code:
```
def logits_to_tokens_loss(logits, tokens):
  # The last "n_digit+1" tokens are the addition answer probabilities
  ans_logits = logits[:, -(n_digits+2):-1]

  # The last "n_digit+1" tokens are the model’s answer.
  ans_tokens = tokens[:, -(n_digits+1):]
```

### Issue:

- The comment indicates both `ans_logits` and `ans_tokens` should refer to the last "n_digit+1" tokens.

- However, `ans_logits` uses `logits[:, -(n_digits+2):-1]`, while `ans_tokens` uses `tokens[:, -(n_digits+1):]`.

### Question:

Why is there a difference in indexing between `ans_logits` and `ans_tokens` when the comments suggest they should be the same?

**Comment by PhilipQuirke** [comment]
Thank you for your comment. The previous code was correct but the comments were not. I've improved the code and comments for clarity. The key bits now read:

````
def logits_to_tokens_loss(logits, tokens):
  # Adding 2 five-digit numbers gives a six-digit answer
  n_answer_digits = n_digits+1

  # The addition answer digit token probabilities
  # The "+1" below is needed because each answer digit calculations occurs one token before that answer digit's token is revealed.
  ans_logits = logits[:, -(n_answer_digits+1):-1]

  # The addition answer digit tokens
  ans_tokens = tokens[:, -(n_answer_digits):]
````
